roles/StackStorm.st2/tasks/main.yml

```diff
@@ -142,6 +141,14 @@
   loop: "{{ st2_services }}"
   tags: st2
 
+- name: Modify st2ctl to only control the given services
+  become: yes
+  lineinfile:
+    path: /opt/stackstorm/st2/bin/st2ctl
+    regexp: '^COMPONENTS='
+    line: "COMPONENTS=\"{{ st2_services | join (' ') }}\""
```

Comment:
Only other thing i can think of is some way of identifying which services are supposed to be enabled on a host (thinking a config file, maybe `st2.conf`).

However, this would require changes to st2 core itself.

Relying on systemd themselves being enabled and grepping that output is unreliable as something or someone maybe have disabled a service that should be enabled.

I think this is a decent middle ground approach until we figure out if/how to support this in st2 core.

Reply:
Check out 
https://github.com/StackStorm/st2/blob/0e8ae756f30ffe2e017c64bff67830abdee7f7c9/st2common/bin/st2ctl#L20-L28